src/Client.js

```diff
@@ -67,6 +67,19 @@ class Client extends EventEmitter {
     async initialize() {
         let [browser, page] = [null, null];
         
+        // eslint-disable-next-line no-useless-escape
+        const foldernameRegex = /^(?!.{256,})(?!(aux|clock\$|con|nul|prn|com[1-9]|lpt[1-9])(?:$|\.))[^ ][ \.\w-$()+=[\];#@~,&amp;']+[^\. ]$/i;
+
+        if (this.id && !foldernameRegex.test(this.id)) throw TypeError('Invalid client ID. Make sure you abide by the folder naming rules of your operating system.');
+
+        const dirPath = path.join(process.cwd(), this.options.dataPath, this.id ? 'session-' + this.id : 'session');
+
+        if (!fs.existsSync(dirPath)) {
+            await Util.createNestedDirectory(dirPath);
```

Comment:
Is this necessary? At least on macOS it doesn't seem like it's needed, and it's actually not creating the directory properly. It seems to be considering absolute paths as relative and creates the directory in the wrong location. 

Puppeteer seems to not have a problem with non-existent directories. Would love to know if windows users have an issue with this though.

Reply:
Hmm, I will test that, thanks!

Reply:
works as said, thanks again